common-content/en/module/logic/search/index.md

```diff
@@ -0,0 +1,119 @@
++++
+title = "1000 Bottles of Wine"
+time = 60
+emoji= "🐀"
+[build]
+render = 'never'
+list = 'local'
+publishResources = false
++++
+
+Now we're going to solve a puzzle together. We're going to think through a few different ways to approach it. This is a well-known puzzle and you may certainly google the answer. However, you will steal from yourself the opportunity of learning this important idea, so maybe _don't_ do that.
+
+{{<note title="A long awaited party" type="activity">}}
+**The King** has stocked his wine cellar with one thousand bottles of wine for his birthday party tomorrow. However, the King's evil brother, the Duke, has poisoned one of the bottles! The poison is lethal and will kill anyone who takes even one sip within one hour. You, the Butler, must find the poisoned bottle and save the party! But how?
+{{</note>}}
+
+> #### Conditions:
+>
+> 🍾 You have **1000** wine bottles, one of which is poisoned.  
+> 🐀 You have **500** rats you can use to test the wine.  
+> ⌛ The poison is lethal and will kill within **1** hour.
+
+{{<tabs name="boolean">}}
+===[[Write down your solution]]===
+
+In your notebook, write down your solution to this problem. Here are the constraints again:
+
+🍾 You have **1000** wine bottles, one of which is poisoned.  
+🐀 You have **500** rats you can use to test the wine.  
+⌛ The poison is lethal and will kill within **1** hour.
+
+===[[🍾 🍾 🐀 Look at a solution]]===
+
+### 🍾 🍾 🐀
+
+One way to think of this is to reduce this problem to its simplest case. If you have 2 bottles and 1 rat, you can test the wine by giving the rat a sip from one bottle. Wait one hour. If the rat is alive, the bottle is safe. If the rat is dead, the bottle is poisoned.
+
+With 500 rats, it's similar, but not the same. Divide the 1000 bottles into 500 groups of 2. In round 1, give each rat a sip from one bottle in each group. Wait one hour. If the rat is alive, the bottle is safe. If the rat is dead, you found the poisoned bottle. If all rats are alive, the poisoned bottle is in the 500 bottles that were not tested. You still have 500 rats, so you can have each rat test the other bottle in their group.
+
+This solution will take a maximum of 2 hours.
+
+{{</tabs>}}
+
+> #### New condition
+>
+> You now have just **10** rats to test the wine.
+
+{{<tabs name="decimation">}}
+===[[Write down your solution]]===
+
+In your notebook, write down your solution to this problem. Here are the constraints again:
+
+🍾 You have **1000** wine bottles, one of which is poisoned.  
+🐀 You have **10** rats you can use to test the wine.  
+⌛ The poison is lethal and will kill within **1** hour.
+
+===[[Look at a solution]]===
+
+With just 10 rats, we need a way to make our problem simpler. Think of it as a search pattern, narrowing down our suspects with each round of testing.
```

Comment:
I'm not sure I'd characterise the improvement as simplifying? We need a more _efficient_ solution maybe?